packages/sdk/src/types/util.ts

```diff
@@ -0,0 +1,163 @@
+/**
+ * Discriminator for {@linkcode TypedJson}. The member used to identify each type is
+ * exported from each type module and can be accessed through named export `JSON_DISCRIMINATOR`.
+ */
+export enum TypedJsonDiscriminator {
```

Comment:
I like this approach over the `_nominal` that I introduced, since the type error message becomes better. :+1: